app/frontend/src/search/ui/stats.js

```diff
@@ -0,0 +1,41 @@
+export const renderStats = (renderOptions, isFirstRender) => {
+  const { nbHits, nbPages, page, hitsPerPage, widgetParams } = renderOptions;
+
+  if (isFirstRender) {
+    return;
+  }
+
+  let results = constructResults(nbHits);
+  let last = constructLastResult(nbPages, page, nbHits, hitsPerPage);
+  let first = constructFirstResult(nbPages, page, hitsPerPage);
+
+  widgetParams.container.innerHTML = `
+    Showing <b>${first}</b> to <b>${last}</b> of <b>${nbHits}</b> ${results}
+  `;
+};
+
+export const constructResults = (results) => {
+  if (results == 1) {
+    return 'result';
+  } else {
+    return 'results';
+  }
+};
+
+export const constructLastResult = (pages, page, results, resultsPerPage) => {
+  if (pages == 0) {
+    return '0';
+  } else if (page + 1 == pages) {
+    return `${results}`;
+  } else {
+    return `${(page + 1) * resultsPerPage}`;
+  }
```

Comment:
if i ever have if/else of 3 or more blocks, i always go for a switch case statement, personal preference maybe, its not a biggie